File: 071.rb
# 071
# Simplify Path
# 2015-12-23
#####################################################
# @param {String} path
# @return {String}
def simplify_path(path)
  v = []
  path.split('/').each do |a|
    if a == '..'
      v.pop
    elsif a != '.' and a != ''
      v.push(a)
    end
  end
  return '/' if v.empty?
  '/'+v.join('/')
end
#####################################################
path = "/..."
r = simplify_path(path)
puts "^"+r+"$"

File: 071.js
//  071
//  Simplify Path
//  2015-12-23
/////////////////////////////////////////////////////
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
    var v = [];
    var dirs = path.split('/');
    for (var i = 0; i < dirs.length; i++) {
        if (dirs[i] == '..')
            v.pop();
        else if (dirs[i] != '.' && dirs[i] != '')
            v.push(dirs[i]);
    }
    if (v.length == 0)
        return '/';
    return '/'+v.join('/');
};

/////////////////////////////////////////////////////
var path = "/.../";
var r = simplifyPath(path);
console.log(r);

